src/Rfc6455Client.php

```diff
@@ -146,7 +146,16 @@ public function getIterator(): \Iterator
 
     public function receive(?Cancellation $cancellation = null): ?Message
     {
-        return $this->messagePipeline->continue($cancellation);
+        $it = $this->messageEmitter->iterate();
```

Comment:
I can't help but feel that this isn't the way you're supposed to do this... it runs completely off messageEmitter rather than messagePipeline.  Not sure is messagePipeline is even needed anymore, or if it could be replaced with the result of `messageEmitter->iterate()` to save ourselves a call here every time?